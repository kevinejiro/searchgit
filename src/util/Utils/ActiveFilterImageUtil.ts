import contributorsImageActive from "../../assets/images/contributorsActive.svg";
import repoActive from "../../assets/images/repoActive.svg";
import contributorsImage from "../../assets/images/contributors.svg";
import repoImage from "../../assets/images/repo.svg";

const ActiveFilterImageUtil = {
  getImageList(input: string) {
    let contributionsImageUrl;
    let repositoriesImageUrl;
    let urlList;
    switch (input) {
      case "user":
        contributionsImageUrl = contributorsImageActive;
        repositoriesImageUrl = repoImage;
        urlList = [contributionsImageUrl, repositoriesImageUrl];
        return urlList;
      case "org":
        contributionsImageUrl = contributorsImage;
        repositoriesImageUrl = repoActive;
        urlList = [contributionsImageUrl, repositoriesImageUrl];
        return urlList;
      default:
        contributionsImageUrl = contributorsImage;
        repositoriesImageUrl = repoImage;
        urlList = [contributionsImageUrl, repositoriesImageUrl];
        return urlList;
    }
  },
};

export default ActiveFilterImageUtil;
