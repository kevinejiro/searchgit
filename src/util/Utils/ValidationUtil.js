const ValidationUtil = {
	nullChecker(input) {
		if (input !== undefined && input !== null) {
			return input;
		}

		return "N/A";
	},
};

export default ValidationUtil;
