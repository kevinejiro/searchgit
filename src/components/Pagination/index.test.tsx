import { render, screen } from "../../util/Utils/test-utils";
import userEvent from "@testing-library/user-event";

import { Pagination, Props } from "./index";
describe("<Pagination />", () => {
  it("renders the component from the beginning", () => {
    const mockProps: Props = {
      page: 1,
      totalPages: 5,
      handlePagination: jest.fn(),
    };
    const component = render(<Pagination {...mockProps} />);

    userEvent.click(
      screen.getByRole("button", {
        name: /1/i,
      })
    );
    expect(mockProps.handlePagination).toBeCalledWith(1);

    userEvent.click(
      screen.getByRole("button", {
        name: /2/i,
      })
    );
    expect(mockProps.handlePagination).toBeCalledWith(2);

    userEvent.click(
      screen.getByRole("button", {
        name: /3/i,
      })
    );
    expect(mockProps.handlePagination).toBeCalledWith(3);

    userEvent.click(
      screen.getByRole("button", {
        name: />/i,
      })
    );
    expect(mockProps.handlePagination).toBeCalledWith(2);
    expect(component).toMatchSnapshot();

    // expect(mockProps.handlePagination).toBeCalledWith(2);
    // wrapper.getByRole("button").at(2).simulate("click");
    // expect(mockProps.handlePagination).toBeCalledWith(3);w
    // wrapper.getByRole("button").at(3).simulate("click");
    // expect(mockProps.handlePagination).toBeCalledWith(5);
    // wrapper.getByRole("button").at(4).simulate("click");
    // expect(mockProps.handlePagination).toBeCalledWith(2);
    // expect(wrapper.render()).toMatchSnapshot();
  });
});
