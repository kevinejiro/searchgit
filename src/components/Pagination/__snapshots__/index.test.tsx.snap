// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> renders the component from the beginning 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="pagination"
      >
        <div
          class="paginationWrapper"
        >
          <button
            class="pageItem active"
            type="button"
          >
            1
          </button>
          <button
            class="pageItem"
            type="button"
          >
            2
          </button>
          <button
            class="pageItem"
            type="button"
          >
            3
          </button>
          <div
            class="separator"
          >
            ...
          </div>
          <button
            class="pageItem false"
            type="button"
          >
            5
          </button>
          <button
            class="pageItem sides"
            type="button"
          >
            &gt;
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="pagination"
    >
      <div
        class="paginationWrapper"
      >
        <button
          class="pageItem active"
          type="button"
        >
          1
        </button>
        <button
          class="pageItem"
          type="button"
        >
          2
        </button>
        <button
          class="pageItem"
          type="button"
        >
          3
        </button>
        <div
          class="separator"
        >
          ...
        </div>
        <button
          class="pageItem false"
          type="button"
        >
          5
        </button>
        <button
          class="pageItem sides"
          type="button"
        >
          &gt;
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
